anomalib/models/ganomaly/torch_model.py

```diff
@@ -363,6 +394,26 @@ def forward(self, batch: Tensor) -> Tensor:
         Returns:
             Tensor: Regeneration scores.
         """
+        padded_batch = self.get_padded_tensor(batch)
         self.generator.eval()
-        _, latent_i, latent_o = self.generator(batch)
+        _, latent_i, latent_o = self.generator(padded_batch)
         return torch.mean(torch.pow((latent_i - latent_o), 2), dim=1).view(-1)  # convert nx1x1 to n
+
+    def get_padded_tensor(self, batch: Tensor) -> Tensor:
```

Comment:
I'll update it in a bit. Static is a good idea. In that case would it make sense to move it to `data/utils/image.py`?

Reply:
Yes, I think it would make sense to move it to a shared location so that other modules could access it as well. In that case we should maybe name it `pad_nextpow2` or similar